examples/ping/ping.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"flag"
+	"fmt"
+	"time"
+
+	"log"
+
+	gorpc "github.com/hsanjuan/go-libp2p-gorpc"
+	libp2p "github.com/libp2p/go-libp2p"
+	host "github.com/libp2p/go-libp2p-host"
+	peerstore "github.com/libp2p/go-libp2p-peerstore"
+	protocol "github.com/libp2p/go-libp2p-protocol"
+	multiaddr "github.com/multiformats/go-multiaddr"
+)
+
+type PingArgs struct {
+	Data []byte
+}
+type PingReply struct {
+	Data []byte
+}
+type PingService struct{}
+
+func (t *PingService) Ping(argType PingArgs, replyType *PingReply) error {
+	log.Println("Received a Ping call")
+	replyType.Data = argType.Data
+	return nil
+}
+
+func createPeer(listenAddr string) host.Host {
+	ctx, _ := context.WithCancel(context.Background())
+	// defer cancel()
+
+	// Create a new libp2p host
+	h, err := libp2p.New(ctx, libp2p.ListenAddrStrings(listenAddr))
+	if err != nil {
+		panic(err)
+	}
+	return h
+}
+
+var protocolID = protocol.ID("/p2p/rpc/ping")
+
+func startServer() {
+	log.Println("Launching host")
+	host := createPeer("/ip4/0.0.0.0/tcp/9000")
+
+	log.Printf("Hello World, my hosts ID is %s\n", host.ID().Pretty())
+	for _, addr := range host.Addrs() {
+		ipfsAddr, err := multiaddr.NewMultiaddr("/ipfs/" + host.ID().Pretty())
+		if err != nil {
+			panic(err)
+		}
+		peerAddr := addr.Encapsulate(ipfsAddr)
+		log.Printf("I'm listening on %s\n", peerAddr)
+	}
+
+	rpcHost := gorpc.NewServer(host, protocolID)
+
+	svc := PingService{}
+	err := rpcHost.Register(&svc)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println("Done")
+
+	for {
+		time.Sleep(time.Second * 1)
+	}
+}
+
+func startClient(host string) {
+	log.Println("Launching client")
+	client := createPeer("/ip4/0.0.0.0/tcp/9001")
+	log.Printf("Hello World, my hosts ID is %s\n", client.ID().Pretty())
+	// for _, addr := range client.Addrs() {
+	// 	log.Printf("I'm listening on %s\n", addr)
+	// }
+	ma, err := multiaddr.NewMultiaddr(host)
+	if err != nil {
+		panic(err)
+	}
+	peerInfo, err := peerstore.InfoFromP2pAddr(ma)
+	if err != nil {
+		panic(err)
+	}
+	ctx := context.Background()
+	err = client.Connect(ctx, *peerInfo)
+	if err != nil {
+		panic(err)
+	}
+	rpcClient := gorpc.NewClient(client, protocolID)
+	numCalls := 0
+	durations := []time.Duration{}
+	betweenPingsSleep := time.Second * 1
+
+	for numCalls < 10 {
+		var reply PingReply
+		var args PingArgs
+
+		// c := 1024 * 100
```

Comment:
Remove commented code.

Reply:
Removed and made into a argument instead. Thanks!